pydsa/kmp.py

```diff
@@ -0,0 +1,39 @@
+# Knutha Morrisa Pratt (KMP) for string matching
+# Complexity: O(n)
+# Reference used: http://code.activestate.com/recipes/117214/
+
+
+def kmp(full_string, pattern):
+    '''
+    It takes two arguments, the main string and the pattern which is to be
+    searched and returns the first position or the starting index of the
+    pattern in the main string (if exists)
```

Comment:
What does it return if pattern isn't found? It should return `-1` when pattern not found, and same should me mentioned here.